paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/ManageScreenInteractor.kt

```diff
@@ -1,31 +1,89 @@
 package com.stripe.android.paymentsheet.verticalmode
 
+import com.stripe.android.lpmfoundations.paymentmethod.PaymentMethodMetadata
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.PaymentMethodCode
 import com.stripe.android.paymentsheet.DisplayableSavedPaymentMethod
+import com.stripe.android.paymentsheet.model.PaymentSelection
 import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel
 import com.stripe.android.ui.core.cbc.CardBrandChoiceEligibility
 import com.stripe.android.uicore.utils.combineAsStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import org.jetbrains.annotations.VisibleForTesting
 
 internal interface ManageScreenInteractor {
     val state: StateFlow<State>
 
+    fun handleViewAction(viewAction: ViewAction)
+
     data class State(
         val paymentMethods: List<DisplayableSavedPaymentMethod>,
+        val currentSelection: DisplayableSavedPaymentMethod?,
     )
+
+    sealed class ViewAction {
+        data class SelectPaymentMethod(val paymentMethod: DisplayableSavedPaymentMethod) : ViewAction()
+    }
 }
 
-internal class DefaultManageScreenInteractor(viewModel: BaseSheetViewModel) : ManageScreenInteractor {
+internal class DefaultManageScreenInteractor(val viewModel: BaseSheetViewModel) : ManageScreenInteractor {
     override val state = combineAsStateFlow(
         viewModel.paymentMethods,
-        viewModel.paymentMethodMetadata
-    ) { paymentMethods, paymentMethodMetadata ->
-        val displayablePaymentMethods = paymentMethods?.map {
-            DisplayableSavedPaymentMethod(
-                displayName = viewModel.providePaymentMethodName(it.type?.code),
-                paymentMethod = it,
-                isCbcEligible = paymentMethodMetadata?.cbcEligibility is CardBrandChoiceEligibility.Eligible,
-            )
+        viewModel.paymentMethodMetadata,
+        viewModel.selection,
+    ) { paymentMethods, paymentMethodMetadata, paymentSelection ->
+        computeInitialState(
+            paymentMethods,
+            paymentMethodMetadata,
+            paymentSelection,
+            viewModel::providePaymentMethodName
+        )
+    }
+
+    override fun handleViewAction(viewAction: ManageScreenInteractor.ViewAction) {
+        when (viewAction) {
+            is ManageScreenInteractor.ViewAction.SelectPaymentMethod ->
+                handlePaymentMethodSelected(viewAction.paymentMethod)
+        }
+    }
+
+    private fun handlePaymentMethodSelected(paymentMethod: DisplayableSavedPaymentMethod) {
+        viewModel.handlePaymentMethodSelected(PaymentSelection.Saved(paymentMethod.paymentMethod))
+        viewModel.handleBackPressed()
```

Comment:
we are supposed to go back to the home screen per the figma. I'm not sure it's the best flow, but it's trivial to remove and we should discuss during design bash